ChangeSet
  1.1799 04/07/08 12:47:40 olh@suse.de +1 -0
  [PATCH] USB: fix lockup with 2.6 keyspan_pda driver
  
  I suggest to put the pointer to port instead of the pointer to the
  pointer to port to the workqueue data.
  Did that driver ever work with 2.6?
  
  
  Signed-off-by: Greg Kroah-Hartman <greg@kroah.com>

  drivers/usb/serial/keyspan_pda.c
    1.37 04/07/08 03:26:38 olh@suse.de[greg] +1 -1
    USB: fix lockup with 2.6 keyspan_pda driver

.........................................................................
diff -p -Nru a/drivers/usb/serial/keyspan_pda.c b/drivers/usb/serial/keyspan_pda.c
--- a/drivers/usb/serial/keyspan_pda.c	Fri Jul  9 11:32:33 2004
+++ b/drivers/usb/serial/keyspan_pda.c	Fri Jul  9 11:32:33 2004
@@ -781,7 +781,7 @@ static int keyspan_pda_startup (struct u
 	usb_set_serial_port_data(serial->port[0], priv);
 	init_waitqueue_head(&serial->port[0]->write_wait);
 	INIT_WORK(&priv->wakeup_work, (void *)keyspan_pda_wakeup_write,
-			(void *)(&serial->port[0]));
+			(void *)(serial->port[0]));
 	INIT_WORK(&priv->unthrottle_work,
 			(void *)keyspan_pda_request_unthrottle,
 			(void *)(serial));
.........................................................................
# vim: syntax=diff

